classes/local/table/files_table.php

```diff
@@ -113,11 +113,19 @@ public function col_link(\stdClass $row) {
         global $DB;
 
         if (substr($row->component, 0, strlen('mod_')) === "mod_") {
-            list ($course, $cm) = get_course_and_cm_from_cmid($row->instanceid);
-            if (!empty($cm)) {
-                $url = new \moodle_url($cm->url);
+            switch ((int)$row->contextlevel) {
+                case CONTEXT_MODULE:
+                    list ($course, $cm) = get_course_and_cm_from_cmid($row->instanceid);
+                    if (!empty($cm)) {
+                        $url = new \moodle_url($cm->url);
+                    }
+                    break;
+                case CONTEXT_COURSE:
```

Comment:
how could it be that $row->component contains mod, but $row->contextlevel would be CONTEXT_COURSE?

Reply:
Hi, there are some module type components where the context level is course or even system, for instance:

```
Component: mod_hvp
filearea: exports
context: 50
```

This record (mdl_files) is created each time a user creates a hvp activity with the option `Allow download` enabled; or when a user downloads a hvp activity if the previous record does not exits. The context in this case is 50 (course).

There are other module type components in mdl_files where the contextlevel is not CONTEXT_MODULE:


 

```
select distinct f.component, f.filearea, ctx.contextlevel
from mdl_context ctx
join mdl_files f ON f.contextid = ctx.id
where f.component like 'mod_%' and contextlevel <> 70;

e.g:

component,filearea,contextlevel
mod_turnitintool,submission,30
mod_data,site_presets,10
mod_hvp,exports,50
mod_hvp,libraries,10
mod_hvp,editor,50
mod_hvp,cachedassets,10
mod_hvp,library_archives,10
```